Add syntax verification to Travis

Hi,

I installed vitasdk with vdpm on my linux machine and ran into the following error which is fixed by this pull request #339 .
```
Scanning dependencies of target kernel
[ 16%] Building C object CMakeFiles/kernel.dir/main.c.obj
In file included from /home/philipp/Projects/VitaShell/modules/kernel/main.c:21:0:
/home/philipp/Projects/vitasdk/arm-vita-eabi/include/psp2kern/kernel/sysmem.h:446:41: error: unknown type name 'kernel_message_ctx'
 int ksceDebugPrintf2(int msg_type_flag, kernel_message_ctx *msg_ctx, const char *fmt, ...);
```
Could you add travis-ci checks to prevent merges of broken header files like this?

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,6 +18,9 @@ script:
     - bash $TRAVIS_BUILD_DIR/.travis.d/download_sdk.sh
     - export VITASDK=$PWD/vitasdk
     - export PATH=$VITASDK/bin:$PATH
+    - export INCLUDE_DIR=$TRAVIS_BUILD_DIR/include
+    - arm-vita-eabi-gcc -I$INCLUDE_DIR -c $INCLUDE_DIR/vitasdk{,kern}.h
+    - arm-vita-eabi-g++ -I$INCLUDE_DIR -c $INCLUDE_DIR/vitasdk{,kern}.h
     - bash $TRAVIS_BUILD_DIR/.travis.d/download_external_libs.sh
     - cd $VITASDK/arm-vita-eabi/include
     - rm -rf psp2 psp2kern vitasdk vitasdk.h

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,11 @@ script:
     - export VITASDK=$PWD/vitasdk
     - export PATH=$VITASDK/bin:$PATH
     - export INCLUDE_DIR=$TRAVIS_BUILD_DIR/include
+    # Make sure headers are valid
     - arm-vita-eabi-gcc -I$INCLUDE_DIR -c $INCLUDE_DIR/vitasdk{,kern}.h
     - arm-vita-eabi-g++ -I$INCLUDE_DIR -c $INCLUDE_DIR/vitasdk{,kern}.h
+    # Remove compiled headers
+    - find $INCLUDE_DIR -name "*.gch" -type f -delete
     - bash $TRAVIS_BUILD_DIR/.travis.d/download_external_libs.sh
     - cd $VITASDK/arm-vita-eabi/include
     - rm -rf psp2 psp2kern vitasdk vitasdk.h

--- a/include/psp2kern/kernel/cpu.h
+++ b/include/psp2kern/kernel/cpu.h
@@ -34,6 +34,18 @@ extern "C" {
 		asm volatile ("mcr p15, 0, %0, c13, c0, 3" :: "r" (state) : "memory"); \
 	} while (0)
 
+
+/**
+ * @brief      Writeback a range of L1 dcache (without L2)
+ *
+ * @param      ptr   The pointer
+ * @param[in]  len   The length
+ *
+ * @return     Zero on success
+ */
+int ksceKernelCpuDcacheWritebackRange(const void *ptr, size_t len);
+
+
 /**
  * @brief      Save process context
  *
@@ -88,8 +100,10 @@ static inline int ksceKernelCpuUnrestrictedMemcpy(void *dst, const void *src, si
 	asm ("mrc p15, 0, %0, c3, c0, 0" : "=r" (prev_dacr));
 	asm ("mcr p15, 0, %0, c3, c0, 0" :: "r" (0xFFFF0000));
 
-	memcpy(dst, src, len);
-	ksceKernelCpuDcacheWritebackRange((uintptr_t)dst & ~0x1F, (len + 0x1F) & ~0x1F);
+	for (size_t i=0; i < len; i++) {
+		((char *) dst)[i] = ((char *) src)[i];
+	}
+	ksceKernelCpuDcacheWritebackRange((void *)((uintptr_t)dst & ~0x1F), (len + 0x1F) & ~0x1F);
 
 	asm ("mcr p15, 0, %0, c3, c0, 0" :: "r" (prev_dacr));
 	return 0;
@@ -117,16 +131,6 @@ int ksceKernelCpuDisableInterrupts(void);
  * @return     Zero on success
  */
 int ksceKernelCpuEnableInterrupts(int flags);
-
-/**
- * @brief      Writeback a range of L1 dcache (without L2)
- *
- * @param      ptr   The pointer
- * @param[in]  len   The length
- *
- * @return     Zero on success
- */
-int ksceKernelCpuDcacheWritebackRange(const void *ptr, size_t len);
 
 /**
  * @brief      Invalidate a range of L1 dcache (without L2)

--- a/include/psp2kern/lowio/pervasive.h
+++ b/include/psp2kern/lowio/pervasive.h
@@ -9,7 +9,7 @@
 #include <psp2kern/types.h>
 
 #ifdef __cplusplus
-"C" {
+extern "C" {
 #endif
 
 int kscePervasiveUartClockDisable(int port);

